hassil/recognize.py

```diff
@@ -314,174 +341,175 @@ def _remove_skip_words(text: str, skip_words: Iterable[str]) -> str:
-                    item_context
-                    for group_context in group_contexts
-                    for item_context in match_expression(group_context, item)
-                ]
-                if not group_contexts:
-                    break
-
-            for group_context in group_contexts:
-                yield group_context
+                for group_context in group_contexts:
+                    yield group_context
         else:
             raise ValueError(f"Unexpected sequence type: {seq}")
 
     elif isinstance(expression, ListReference):
         # {list}
         list_ref: ListReference = expression
-        if (not context.slot_lists) or (list_ref.list_name not in context.slot_lists):
+        if (not settings.slot_lists) or (list_ref.list_name not in settings.slot_lists):
             raise MissingListError(f"Missing slot list {{{list_ref.list_name}}}")
 
-        slot_list = context.slot_lists[list_ref.list_name]
-        if isinstance(slot_list, TextSlotList) and context.text:
-            text_list: TextSlotList = slot_list
-            # Any value may match
-            for slot_value in text_list.values:
-                value_contexts = match_expression(
-                    MatchContext(
-                        # Copy over
-                        text=context.text,
-                        slot_lists=context.slot_lists,
-                        expansion_rules=context.expansion_rules,
-                        entities=context.entities,
-                        intent_context=context.intent_context,
-                    ),
-                    slot_value.text_in,
-                )
+        slot_list = settings.slot_lists[list_ref.list_name]
+        if isinstance(slot_list, TextSlotList):
+            if context.text:
+                text_list: TextSlotList = slot_list
+                # Any value may match
+                for slot_value in text_list.values:
+                    value_contexts = match_expression(
+                        settings,
+                        MatchContext(
+                            # Copy over
+                            text=context.text,
+                            entities=context.entities,
+                            intent_context=context.intent_context,
+                        ),
+                        slot_value.text_in,
+                    )
 
-                for value_context in value_contexts:
-                    entities = context.entities + [
-                        MatchEntity(
-                            name=list_ref.slot_name,
-                            value=slot_value.value_out,
-                            text=context.text[: -len(value_context.text)]
-                            if value_context.text
-                            else context.text,
-                        )
-                    ]
+                    for value_context in value_contexts:
+                        entities = context.entities + [
```

Comment:
Why not reuse the list.?

Reply:
I'm not sure I understand. Each value will produce a different `MatchEntity` (different `text`), so how could I reuse the list?